Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -994,6 +994,29 @@ NiftiImageIO::SetImageIOMetadataFromNIfTI()
   std::ostringstream intent_name;
   intent_name << nim->intent_name;
   EncapsulateMetaData<std::string>(thisDic, "intent_name", intent_name.str());
+
+  // The below were added after ITK 5.3rc2.
+
+  EncapsulateMetaData<float>(thisDic, "qfac", nim->qfac);
+
+  std::vector<float> qto_xyz;
+  qto_xyz.push_back(nim->qto_xyz.m[0][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[0][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[0][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[0][3]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[1][3]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[2][3]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][0]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][1]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][2]);
+  qto_xyz.push_back(nim->qto_xyz.m[3][3]);
+  EncapsulateMetaData<std::vector<float>>(thisDic, "qto_xyz", qto_xyz);
```

Comment:
Encoding `mat44` itself I don't like, because it's a niftilib data structure, which I don't want to leak outside ITK.  Adding an `operator==` to it would also require changing niftilib, and it's C anyway.

Reply:
`mat44` is defined in the global namespace (obviously, because it is C), so technically it is allowed to add an `operator==` to "our" global namespace, for example as follows:

```
bool
operator==(const mat44 & lhs, const mat44 & rhs)
{
  return std::equal(
    std::begin(lhs.m), std::end(lhs.m), std::begin(rhs.m), [](const auto & leftRow, const auto & rightRow) {
      return std::equal(std::begin(leftRow), std::end(leftRow), std::begin(rightRow));
    });
}
```

Another option might be to adjust `itk::MetaDataObject<MetaDataObjectType>` to allow types without an `operator==`.

I realize that it is my fault that `itk::MetaDataObject<MetaDataObjectType>`  requires `operator==` for the type to be stored: pull request #2246 commit 694bbc0a1f58bdb4564252e6f75d8611f4e2a5ed  😮  It could be adjusted to use std::memcmp, for types that don't have an `operator==`.  (@Leengit A SFINAE opportunity!!!)

Reply:
> Encoding mat44 itself I don't like, because it's a niftilib data structure

OK, but then, what would you think about encapsulating the matrix as a `itk::Matrix<float, 4, 4>`?

Reply:
That doesn't make me recoil with C++ whiplash. :)  But it starts to be very different from the rest of the existing code in that function, which seems to prefer basic data structures.

Reply:
The other thing I like about this PR as-is is that's its easy to cherry pick just this change.  For my own app, I'll be cherry-picking this into ITK 5.2 for a hot fix.

Reply:
OK, but do you agree that `itk::Matrix<float, 4, 4>` would be a proper data type to use when storing `nim->qto_xyz` into the `itk::MetaDataDictionary`?

Reply:
The type used here should be one of the explicitly instantiated MetaDataObject types: 
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkMetaDataObject.h#L298-L321

I believe only itk::Matrix<float,3,3> is there, itk::Matrix<float,4,4> would need to be added.